pkg/rhcos/builds.go

```diff
@@ -36,42 +36,79 @@ type metadata struct {
 	OSTreeVersion string `json:"ostree-version"`
 }
 
-func fetchLatestMetadata(ctx context.Context, channel string) (metadata, error) {
-	build := buildName
-	var err error
-	if build == "" {
-		build, err = fetchLatestBuild(ctx, channel)
-		if err != nil {
-			return metadata{}, errors.Wrap(err, "failed to fetch latest build")
+var rhcosBuildOnce sync.Once
+var rhcosBuildID string
+var rhcosBuildChannel string
+
+func fetchRHCOSBuild(ctx context.Context) (metadata, error) {
+	rhcosBuildOnce.Do(func() {
```

Comment:
Why the sync once?

Reply:
Hmm, I had though this could be called multiple times; I think it was in the past but it looks like we only call `func (i *Image) Generate(p asset.Parents) error {` once now?  I'll remove the once.

Reply:
Updated to drop the `sync`.